Rework node cert workflow in cert and deployment docs

jseldess opened the pull request:
After generating and uploading certs to a node, remove
the node cert and key before generating those for the next
node. This is instead of using our custom `--overwrite` flag,
though I still mention the flag as an alternative.

Fixes #1678

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/1691)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/6ba70db1592924854fe3c9c9fac99148faf582d7/

mberhault commented on v1.0/create-security-certificates.md:
```diff
@@ -131,54 +153,48 @@ If you need to troubleshoot this command's behavior, you can change its [logging
+3. Delete the local copy of the first node's certificate and key:
+
+    ~~~ shell
+    $ rm -rf certs/node.crt certs/node.key
```
you can drop the `-rf`. We're definitely not deleting directories (`-r`) and the permissions set by `cockroach cert` means we don't need to force the delete (`-f`).

jseldess replied:
Done.

mberhault commented on v1.0/manual-deployment.md:
```diff
@@ -86,19 +94,29 @@ Locally, you'll need to [create the following certificates and keys](create-secu
 	$ ssh <username>@<node1 address> "mkdir certs"
 
-	# Upload the CA certificate, client (root) certificate and key, and node certificate and key:
+	# Upload the CA certificate, client certificates and keys, and node certificate and key:
```
We don't need to copy the client certificate and key to the node. I can understand `root` being useful for local debugging, but other users really don't need to be there.

jseldess replied:
Done.

mberhault commented on v1.0/create-security-certificates.md:
```diff
@@ -113,12 +116,31 @@ If you need to troubleshoot this command's behavior, you can change its [logging
+    -rw-r--r--  1 maxroach  maxroach  1.1K Jul 10 14:12 ca.crt
     ~~~
 
+### Create the certificate and key pair for a client
```
This was fine as the last step, keeping node and client certs separate makes things a bit clearer.

jseldess replied:
Done.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/4528961bad2cb6d43840673b7aac3bc2667bcaf1/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/81ff4815eb109985874bc12b1e02b3c5d4601556/

mberhault commented on v1.0/create-security-certificates.md:
```diff
@@ -96,106 +96,144 @@ If you need to troubleshoot this command's behavior, you can change its [logging
+    -rw-------  1 maxroach  maxroach  1.6K Jul 10 14:16 node.key
     ~~~
 
 2. Upload the files to the first node.
```
uh. both `upload the files to the Nth node` steps are empty in this doc. Is it because this is an overview of how certs work? It seems odd though.

bdarnell commented:
Review status: 0 of 12 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[v1.0/manual-deployment.md, line 97 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1691#-KogXyR48Dgyfhfahe-1-r1-97:-KoiV9nGuWesbUqdWlJH:bitj7u8) ([raw file](https://github.com/cockroachdb/docs/blob/6ba70db1592924854fe3c9c9fac99148faf582d7/v1.0/manual-deployment.md#L97)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Done.
</blockquote></details>

We may want to mention that copying the root cert is optional (and I'd consider it a bad practice, although it's not technically more of a security risk than the node certificates that are already there)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1691)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[v1.0/manual-deployment.md, line 97 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1691#-KogXyR48Dgyfhfahe-1-r1-97:-KolnY6dUOy--3avpvUk:b-o7rizt) ([raw file](https://github.com/cockroachdb/docs/blob/6ba70db1592924854fe3c9c9fac99148faf582d7/v1.0/manual-deployment.md#L97)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

We may want to mention that copying the root cert is optional (and I'd consider it a bad practice, although it's not technically more of a security risk than the node certificates that are already there)
</blockquote></details>

Since this is considered bad practice, I updated all deployment tutorials to not copy the root user's cert/key to nodes and added a note about doing that if node-local debugging is needed. 

Also updated the "test a cluster" instructions to start the sql client locally instead of on nodes.

Hope these changes are acceptable, @bdarnell. Please let me know.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1691)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[v1.0/create-security-certificates.md, line 157 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1691#-KogXyR48Dgyfhfahdzx-r3-157:-Kolo-rj2mzjEBU-ZxzQ:b-13swmu) ([raw file](https://github.com/cockroachdb/docs/blob/81ff4815eb109985874bc12b1e02b3c5d4601556/v1.0/create-security-certificates.md#L157)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary><blockquote>

uh. both `upload the files to the Nth node` steps are empty in this doc. Is it because this is an overview of how certs work? It seems odd though.
</blockquote></details>

Added those details

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1691)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/2b8ddd5140b8a81b2adad3a1ee4510dd42e3c10f/

bdarnell commented:
Review status: 0 of 12 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[v1.0/manual-deployment.md, line 97 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1691#-KogXyR48Dgyfhfahe-1-r1-97:-KomGQMO5FsIeoHjQRD9:b49opeb) ([raw file](https://github.com/cockroachdb/docs/blob/6ba70db1592924854fe3c9c9fac99148faf582d7/v1.0/manual-deployment.md#L97)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Since this is considered bad practice, I updated all deployment tutorials to not copy the root user's cert/key to nodes and added a note about doing that if node-local debugging is needed. 

Also updated the "test a cluster" instructions to start the sql client locally instead of on nodes.

Hope these changes are acceptable, @bdarnell. Please let me know.
</blockquote></details>

LGTM (but does this break any of our other instructions? If we're relying on ssh'ing to nodes and running cockroach commands then we may want to leave it in for now)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1691)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[v1.0/manual-deployment.md, line 97 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1691#-KogXyR48Dgyfhfahe-1-r1-97:-KomHw8vBHGyigTITg1p:bvj7qxl) ([raw file](https://github.com/cockroachdb/docs/blob/6ba70db1592924854fe3c9c9fac99148faf582d7/v1.0/manual-deployment.md#L97)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

LGTM (but does this break any of our other instructions? If we're relying on ssh'ing to nodes and running cockroach commands then we may want to leave it in for now)
</blockquote></details>

Ah, there is still one place in the manual deployment guide where the root cert is needed on the server: When running `cockroach gen haproxy` to generate the haproxy config. I'll revise the instructions to generate that file locally and copy it to the haproxy server instead. 

Otherwise, I'm pretty sure all cockroach client commands are run locally, not on nodes, in these guides.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1691)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/37c53529cf54d0a11929c77bcab9305b83249b9f/
